vador/src/main/java/com/salesforce/vador/annotation/FutureDateField.kt

```diff
@@ -0,0 +1,25 @@
+package com.salesforce.vador.annotation
+
+import com.salesforce.vador.types.ValidatorAnnotation1
+import java.lang.reflect.Field
+import java.time.LocalDate
+
+@Retention(AnnotationRetention.RUNTIME)
+@Target(AnnotationTarget.FIELD)
+annotation class FutureDateField(val failureKey: String)
+
+class FutureDateFieldValidator<FailureT> : ValidatorAnnotation1<LocalDate, FailureT> {
+  override fun validate(
+    field: Field,
+    value: LocalDate,
```

Comment:
We should support the deprecated [Java 8 Date](https://docs.oracle.com/javase/8/docs/api/java/util/Date.html) also as most of core and Connect APIs still use it.